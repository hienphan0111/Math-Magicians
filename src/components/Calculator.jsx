import React from 'react';
import './Calculator.scss';

const Calculator = () => (
  <div className="cal-container">
    <span id="result">0</span>
    <span id="ac">AC</span>
    <span id="plus-minus">&#177;</span>
    <span id="mod">%</span>
    <span id="divide">&#247;</span>
    <span id="multi">x</span>
    <span id="minus">-</span>
    <span id="plus">+</span>
    <span id="equal">=</span>
    <span id="9">9</span>
    <span id="8">8</span>
    <span id="7">7</span>
    <span id="6">6</span>
    <span id="5">5</span>
    <span id="4">4</span>
    <span id="3">3</span>
    <span id="2">2</span>
    <span id="1">1</span>
    <span id="zero">0</span>
    <span id="dot">.</span>
  </div>
);

export default Calculator;
