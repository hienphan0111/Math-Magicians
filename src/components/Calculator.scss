.cal-container {
  width: 400px;
  height: 500px;
  margin: 10px auto;
  display: grid;
  grid-template-columns: repeat(4, 1fr);
  grid-template-rows: repeat(6, 1fr);
  justify-content: center;
  align-items: center;
  border: 1px solid #000;
  cursor: pointer;
  font-size: 18px;
  font-weight: 500;

  #result {
    grid-column-start: 1;
    grid-column-end: span 4;
    background-color: rgba(128, 128, 128, 0.567);
    height: 100%;
    display: flex;
    justify-content: flex-end;
    padding-right: 10px;
    font-size: 20px;
  }

  #multi {
    grid-column-start: 4;
    grid-row-start: 3;
  }

  #plus {
    grid-column-start: 4;
    grid-row-start: 4;
  }

  #minus {
    grid-column-start: 4;
    grid-row-start: 5;
  }

  #equal {
    grid-column-start: 4;
    grid-row-start: 6;
  }

  #zero {
    grid-column-start: 1;
    grid-row-start: 6;
    grid-column-end: span 2;
  }

  #one {
    grid-column-start: 1;
    grid-row-start: 5;
  }

  #two {
    grid-column-start: 2;
    grid-row-start: 5;
  }

  #four {
    grid-column-start: 1;
    grid-row-start: 4;
  }

  
  #five {
    grid-column-start: 2;
    grid-row-start: 4;
  }

  #seven {
    grid-column-start: 1;
    grid-row-start: 3;
  }

  #eight {
    grid-column-start: 2;
    grid-row-start: 3;
  }

  .orange {
    background-color: orange;
    border: 0.7px solid #000;
    height: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
  }

}

.border-center {
  border: 0.5px solid #000;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}
