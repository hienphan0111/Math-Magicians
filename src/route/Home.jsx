import './Home.scss';

const Home = () => (
  <div className="home-container">
    <h2>Welcome to our page</h2>
    <p>
      The Calculator app is a modern Web app written in React
      and typical langues that used to for many web apps.
      The app provides standard calculator functionality.
    </p>
    <p>This app aim to learn how to create an app using React framework</p>
  </div>
);

export default Home;
