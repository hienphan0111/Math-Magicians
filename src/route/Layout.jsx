import { Outlet } from 'react-router-dom';
import Navbar from 'components/Navbar';
import './Layout.scss';

const Layout = () => (
  <div className="main-container">
    <Navbar />
    <Outlet />
  </div>
);

export default Layout;
