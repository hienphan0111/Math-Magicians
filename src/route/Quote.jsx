import './Quote.scss';

const quotes = [
  {
    author: 'Deepak Chopra',
    quote: 'Mathematics expresses values that reflect the cosmos,including orderliness, balance, harmony, logic, and abstract beauty.',
  },
  {
    author: 'G.K. Chesterton',
    quote: 'The difference between the poet and the mathematician is that the poet tries to get his head into the heavens while the mathematician tries to get the heavens into his head.',
  },
  {
    author: 'Malcolm X, The Autobiography of Malcolm X',
    quote: 'I\'m sorry to say that the subject I most disliked was mathematics. I have thought about it. I think the reason was that mathematics leaves no room for argument. If you made a mistake, that was all there was to it.',
  },
  {
    author: 'Stephenie Meyer',
    quote: 'My first feeling was that there was no way to continue. Writing isn\'t like math;in math, two plus two always equals four no matter what your mood is like. With writing, the way you feel changes everything.',
  },
  {
    author: 'Roman Payne',
    quote: 'The ‘Muse’ is not an artistic mystery, but a mathematical equation. The gift are those ideas you think of as you drift to sleep. The giver is that one you think of when you first awake.',
  },
];

const Quote = () => {
  const num = Math.floor(Math.random() * quotes.length);
  const { author, quote } = quotes[num];

  return (
    <div className="quote-container">
      <p>
        &quot;
        {quote}
        &quot;
      </p>
      <span>{author}</span>
    </div>
  );
};

export default Quote;
