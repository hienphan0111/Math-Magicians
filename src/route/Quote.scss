@import url('https://fonts.googleapis.com/css2?family=Lobster&display=swap');

.quote-container {
  max-width: 480px;
  margin: 20px auto;
  font-family: 'Lobster', cursive;
  font-size: 28px;

  span {
    display: flex;
    justify-content: right;
    font-family: 'Franklin Gothic Medium', 'Arial Narrow', Arial, sans-serif;
  }
}
