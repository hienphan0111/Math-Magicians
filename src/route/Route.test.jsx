import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import Home from './Home';
import Layout from './Layout';
import Quote from './Quote';
import Calculator from 'components/Calculator';

describe('Calculator component Snapshot testing suite', () => {
  it('Matches Home DOM Snapshot', () => {
    const component = renderer.create(<Home />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Matches Layout DOM Snapshot', () => {
    const component = renderer.create(
    <MemoryRouter>
      <Calculator />
    </MemoryRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Matches Layout DOM Snapshot', () => {
    const component = renderer.create(
    <MemoryRouter>
      <Layout />
    </MemoryRouter>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Matches Quote DOM Snapshot', () => {
    const component = renderer.create(<Quote />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});