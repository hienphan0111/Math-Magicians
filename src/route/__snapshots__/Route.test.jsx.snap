// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component Snapshot testing suite Matches Home DOM Snapshot 1`] = `
<div
  className="home-container"
>
  <h2>
    Welcome to our page
  </h2>
  <p>
    The Calculator app is a modern Web app written in React and typical langues that used to for many web apps. The app provides standard calculator functionality.
  </p>
  <p>
    This app aim to learn how to create an app using React framework
  </p>
</div>
`;

exports[`Calculator component Snapshot testing suite Matches Layout DOM Snapshot 1`] = `
<div
  className="cal-container"
>
  <span
    className="border-center"
    id="result"
  >
    0
  </span>
  <span
    className="border-center"
    id="ac"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    AC
  </span>
  <span
    className="border-center"
    id="plus-minus"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    +/-
  </span>
  <span
    className="border-center"
    id="mod"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    %
  </span>
  <span
    className="border-center"
    id="divide"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    รท
  </span>
  <span
    className="border-center"
    id="multi"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    x
  </span>
  <span
    className="border-center"
    id="plus"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    +
  </span>
  <span
    className="border-center"
    id="minus"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    -
  </span>
  <span
    className="border-center"
    id="equal"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    =
  </span>
  <span
    className="border-center"
    id="dot"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    .
  </span>
  <span
    className="border-center"
    id="zero"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    0
  </span>
  <span
    className="border-center"
    id="one"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    1
  </span>
  <span
    className="border-center"
    id="two"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    2
  </span>
  <span
    className="border-center"
    id="three"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    3
  </span>
  <span
    className="border-center"
    id="four"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    4
  </span>
  <span
    className="border-center"
    id="five"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    5
  </span>
  <span
    className="border-center"
    id="six"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    6
  </span>
  <span
    className="border-center"
    id="seven"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    7
  </span>
  <span
    className="border-center"
    id="eight"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    8
  </span>
  <span
    className="border-center"
    id="nine"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    tabIndex={0}
  >
    9
  </span>
</div>
`;

exports[`Calculator component Snapshot testing suite Matches Layout DOM Snapshot 2`] = `
<div
  className="main-container"
>
  <nav
    className="navbar"
  >
    <h1>
      Math Magicians
    </h1>
    <ul>
      <li>
        <a
          className="item"
          href="/Math-magicians"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className="item"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className="item last"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Calculator component Snapshot testing suite Matches Quote DOM Snapshot 1`] = `
<div
  className="quote-container"
>
  <p>
    "
    Mathematics expresses values that reflect the cosmos,including orderliness, balance, harmony, logic, and abstract beauty.
    "
  </p>
  <span>
    Deepak Chopra
  </span>
</div>
`;
